src/main/java/com/deloitte/SpaceStation/reservation/service/ReservationServiceImpl.java

```diff
@@ -77,4 +79,23 @@ public void deleteById(Long reservationId) {
         reservationRepository.deleteById(reservationId);
     }
 
+    @Override
+    public ReservationResponseDto putReservation(Long id, ReservationRequestDto reservationRequestDto) {
+        checkIfWorksiteIsAvailable(reservationRequestDto.getWorksiteId(), reservationRequestDto.getStartDate(),
+                reservationRequestDto.getEndDate());
+
+        Reservation reservation = reservationRepository.findById(id)
+                .orElseThrow(() -> new SpaceStationException(Error.RESERVATION_NOT_FOUND));
+
+        reservation.setOwner(userRepository.findById(reservationRequestDto.getOwnerId())
+                .stream().findFirst().orElseThrow(() -> new SpaceStationException(Error.USER_NOT_FOUND)));
+        reservation.setStartDate(reservationRequestDto.getStartDate());
+        reservation.setEndDate(reservationRequestDto.getEndDate());
+        reservation.setWorksite(worksiteRepository.findById(reservationRequestDto.getWorksiteId())
+                .stream().findFirst().orElseThrow(() -> new SpaceStationException(Error.WORKSITE_NOT_FOUND)));
```

Comment:
Tutaj można użyć ReservationMappera do zampowania requesta na reservation - tam już jest w sumie obsługa błędów i rzucanie wyjątków, jeśli nie zostanie coś znalezione

Reply:
Poprawione (na chyba to o co chodziło)